source/common/network/apple_dns_impl.cc

```diff
@@ -79,6 +79,7 @@ ActiveDnsQuery* AppleDnsResolverImpl::resolve(const std::string& dns_name,
     if (error != kDNSServiceErr_NoError) {
       ENVOY_LOG(warn, "DNS resolver error ({}) in dnsServiceGetAddrInfo for {}", error, dns_name);
       chargeGetAddrInfoErrorStats(error);
+      callback(DnsResolver::ResolutionStatus::Failure, {});
```

Comment:
Per slack, can you refactor this function so it's less fragile? Perhaps make this function a `resolveWorker()` and then have it return a tuple of error and active query (to account for the immediate success case) and then raise the error there?

Also, did you audit everywhere else to make sure we are raising errors in all failure cases?

Reply:
Sure. I'm not sure what error we would return since all we do is is return an empty list, but I can put something together.

Let me see if I can add some asserts that verify that we either trigger a callback or cancel the resolution, I think that would cover us well sine we're dealing with the immediate failures with the current change.